Enforce CC and CXX on Clang modern

### Description of Problem, Request, or Question

When building darknet with conanio/clang13-ubuntu16.04 I have:

```
no-unused-result -Wno-unknown-pragmas -Wfatal-errors -DOPENCV -c ./src/image_opencv.cpp -o obj/image_opencv.o
/bin/sh: 1: g++: not found
Makefile:79: recipe for target 'obj/image_opencv.o' failed
make: *** [obj/image_opencv.o] Error 127
make: *** Waiting for unfinished jobs....
darknet/cci.20180914: 
darknet/cci.20180914: ERROR: Package '2defd39034010001348534af454bca19c8622330' build failed
darknet/cci.20180914: WARN: Build folder /home/conan/.conan/data/darknet/cci.20180914/_/_/build/2defd39034010001348534af454bca19c8622330
ERROR: darknet/cci.20180914: Error in build() method, line 84
	env_build.make(args=args)
	ConanException: Error 2 while executing make 'OPENCV=1' -j4
```

It looks for g++, not clang++. Why? Actually it looks for CXX and CC env vars, but are not there, so by default it tries to find g++. It's not first time we have this case, that's why old images enforce CC and CXX.

--- a/modern/clang/Dockerfile
+++ b/modern/clang/Dockerfile
@@ -138,6 +138,11 @@ ARG DISTRO
 COPY --from=builder /tmp/install /tmp/clang
 COPY --from=libstdcpp /usr/local /tmp/gcc
 
+ENV CC=clang \
+    CXX=clang++ \
+    CMAKE_C_COMPILER=clang \
+    CMAKE_CXX_COMPILER=clang++
+
 RUN sudo mv /tmp/gcc/lib64 /usr/local/ \
     && sudo ln -s -f /usr/local/lib64/libstdc++.so.6.0.${LIBSTDCPP_PATCH_VERSION} /usr/local/lib64/libstdc++.so.6 \
     && sudo ln -s -f /usr/local/lib64/libstdc++.so.6 /usr/local/lib64/libstdc++.so \

